hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedRawSSTFileReader.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdds.utils.db.managed;
+
+import org.apache.hadoop.hdds.utils.NativeLibraryLoader;
+import org.apache.hadoop.hdds.utils.NativeLibraryNotLoadedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.util.function.Function;
+
+import static org.apache.hadoop.hdds.utils.NativeConstants.ROCKS_TOOLS_NATIVE_LIBRARY_NAME;
+
+/**
+ * JNI for RocksDB RawSSTFileReader.
+ */
+public class ManagedRawSSTFileReader<T> implements Closeable {
+
+  public static boolean loadLibrary() throws NativeLibraryNotLoadedException {
+    ManagedRocksObjectUtils.loadRocksDBLibrary();
+    if (!NativeLibraryLoader.getInstance()
+        .loadLibrary(ROCKS_TOOLS_NATIVE_LIBRARY_NAME)) {
+      throw new NativeLibraryNotLoadedException(
+          ROCKS_TOOLS_NATIVE_LIBRARY_NAME);
+    }
+    return true;
+  }
```

Comment:
done

Reply:
`loadLibrary` is still in `ManagedRawSstFileReader`. Please move it to `SnapshotDiffManager`.

It is responsibility of the SnapshotDiffManager to make sure native-lib is loaded before using optimized SnapDiff.

Reply:
No I believe this particular piece should reside in this particular module since this class cannot run without the library being loaded. It has to be self contained.

Reply:
Even rocksdb does something similar

Reply:
I disagree with you because if `ManagedRawSstFileReader` has some dependency on `NativeLibrary`, it should be injected to it.
Since it is static loading, we can move this code inside the constructor of `ManagedRawSstFileReader` or static block (which I don't think possible because it is needed to be loaded based on the config) but loading this way doesn't feel good to me.